### PR TITLE
Minor/update icon

### LINKED ISSUE #33: Add icon set as configs of Icon/components
We're finding some undesirable effects caused by having our icon set as a font such as undeterminable icon element sizes (due to font properties and values), and so it feels like a signal to explore SVGs instead.

### PR DESCRIPTION
**Pull request by stevemdixon**
### Purpose
This PR resolves #33 by adding our icon set as individual components.
### Things to Note
We might want to consider whether or not these will be components or configs. 😄

**Update:**
Icons will be their "own" component (i.e. `<AddMediaIcon />`) and will take in a `color` and a `size` prop.
### Review
Keen to see how this feels! 👍

### CODE REVIEW

File: Icon/Icons/AddMediaIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <path d="M6.72854847,13 L10.4427219,13 C10.994669,13 11.1806866,12.6353752 10.8595051,12.1855878 L8.08131797,8.29497314 C7.75836623,7.84270671 7.23939818,7.84518568 6.9182166,8.29497314 L5.17096772,10.7418465 L4.89598004,10.3431742 C4.58220845,9.88827357 4.04270416,9.86191756 3.68900632,10.2814689 L1.19456888,13.2403351 C0.837902685,13.6634075 1.00101068,14 1.55225078,14 L6.42019856,14 C6.96793003,14 7.16465526,13.63226 6.85175993,13.1786298 L6.72854847,13 Z" id="Combined-Shape" />
+      <path d="M12.5,3.5 L12.5,2.49047852 C12.5,2.21505737 12.2761424,2 12,2 C11.7319336,2 11.5,2.21959471 11.5,2.49047852 L11.5,3.5 L10.4904785,3.5 C10.2150574,3.5 10,3.72385763 10,4 C10,4.26806641 10.2195947,4.5 10.4904785,4.5 L11.5,4.5 L11.5,5.50952148 C11.5,5.78494263 11.7238576,6 12,6 C12.2680664,6 12.5,5.78040529 12.5,5.50952148 L12.5,4.5 L13.5095215,4.5 C13.7849426,4.5 14,4.27614237 14,4 C14,3.73193359 13.7804053,3.5 13.5095215,3.5 L12.5,3.5 Z M12,8 C9.790861,8 8,6.209139 8,4 C8,1.790861 9.790861,0 12,0 C14.209139,0 16,1.790861 16,4 C16,6.209139 14.209139,8 12,8 Z" id="Combined-Shape" />
+    </g>
+  </Icon>;
+
+AddMediaIcon.propTypes = {
+  ...Icon.PropTypes,
+};
+
```
Comment (hharnisc) [inline Icon/Icons/AddMediaIcon.jsx]: I think we might want to set the default color at this level.

```
AddMediaIcon.defaultProps = {
  color: 'red', // or whichever
};
```

File: Icon/style.css [hunk 2]
```diff
@@ -1,13 +1,108 @@
 @import '../variables.css';
 
+.curious-blue {
```
Comment (hharnisc) [inline Icon/style.css]: just a thought here. what if these were renamed to the JS equivalents `curiousBlue`.

Then you could just add the `color` prop to to classNames like this

```js
const classes = classNames(styles, 'icon', color);
```

The proptypes would ensure that you pass in a valid color 😄

File: Icon/Icons/AnalyticsIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React, { PropTypes } from 'react';
+import Icon from '../../Icon';
+
+const AnalyticsIcon = ({ size }) =>
```
Comment (hharnisc) [inline Icon/Icons/AnalyticsIcon.jsx]: Would probably be good to use the same api for each of these to make them customizable in the same ways.

```
const AnalyticsIcon = ({ size, color }) => {};
```

File: Icon/Icons/AnalyticsIcon.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        <path d="M10,11 L10,5 C10,4.44771525 9.55228475,4 9,4 C8.44771525,4 8,4.44771525 8,5 L8,11 C8,11.5522847 8.44771525,12 9,12 C9.55228475,12 10,11.5522847 10,11 Z" id="bar-2" />
+        <path d="M14,11 L14,3 C14,2.44771525 13.5522847,2 13,2 C12.4477153,2 12,2.44771525 12,3 L12,11 C12,11.5522847 12.4477153,12 13,12 C13.5522847,12 14,11.5522847 14,11 Z" id="bar-3" />
+      </g>
+    </g>
+  </Icon>;
+
+AnalyticsIcon.propTypes = {
+  size: PropTypes.oneOf(['small', 'large']),
+};
+
```
Comment (hharnisc) [inline Icon/Icons/AnalyticsIcon.jsx]: also set the default color here

File: Icon/index.jsx [hunk 5]
```diff
@@ -2,19 +2,83 @@ import React, { PropTypes } from 'react';
...
+    </svg>
+  );
 };
 
-
 Icon.propTypes = {
-  className: PropTypes.string,
+  children: PropTypes.node,
+  color: PropTypes.oneOf([
+    'curious-blue',
```
Comment (hharnisc) [inline Icon/index.jsx]: might be good to set these to their js equivalent ie `curiousBlueLight`

File: Icon/Icons/AddMediaIcon.jsx (hunk 1 above)
Comment (stevemdixon) [inline Icon/Icons/AddMediaIcon.jsx]: Awesome! I'm wondering if it feels smooth to have the CSS property value in here or perhaps somehow extracting the CSS variable out. 🤔

File: Icon/style.css (hunk 2 above)
Comment (stevemdixon) [inline Icon/style.css]: Ah, might we do that for `sizes` also?

File: Icon/Icons/AnalyticsIcon.jsx (hunk 3 above)
Comment (stevemdixon) [inline Icon/Icons/AnalyticsIcon.jsx]: Awesome catch! For sure! 👍

File: Icon/style.css (hunk 2 above)
Comment (hharnisc) [inline Icon/style.css]: I think size would (or should) all be handled by the `Icon` component. I'd expect this to be a pass through.

**Comment by stevemdixon** [comment]
Hey, @hharnisc! I'm curious to see how our color solution feels here. Once this is merged, I'll go ahead and do a full sweep of our components that use icons both here and `buffer-web-components`. 👍

**Pull request merged**